### PR TITLE
[#2] [UI] As a user, when signing in with email for the first time, I can register an account

### LINKED ISSUE #2: [UI] As a user, when signing in with email for the first time, I can register an account
## Why

Users must be authenticated to use the application, so we need to create a signup screen

## Acceptance Criteria

The screen includes:
- Email field
- Password field
- Confirm password field

### PR DESCRIPTION
**Pull request by suho**
https://github.com/suho/google-search-ruby/issues/2

## What happened 👀

Users must be authenticated to use the application, so we need to create a signup screen.
 
## Insight 📝

The register screen from devise includes:
- Email field
- Password field
- Confirm password field

In this pull request, I update the style for the UI from devise and also add logic for the header and the greetings message after registering successfully.
 
## Proof Of Work 📹

| Description | Screenshots|
|-|-|
|Register - Web|web-register|
|Regiter - Mobile|mobile-register|
|Error|error|
|Success|welcome|

### CODE REVIEW

File: app/views/devise/registrations/new.html.erb [hunk 1]
```diff
@@ -1,29 +1,34 @@
...
+    <% end %>
+    <%= f.password_field :password, autocomplete: "new-password", class: "form-control", :required => true, :minlength => 6 %>
   </div>
 
-  <div class="field">
-    <%= f.label :password_confirmation %><br />
-    <%= f.password_field :password_confirmation, autocomplete: "new-password" %>
+  <div class="form-group">
+    <%= f.label :password_confirmation %>
+    <%= f.password_field :password_confirmation, autocomplete: "new-password", class: "form-control", :required => true, :minlength => 6 %>
```
Comment (junan) [inline app/views/devise/registrations/new.html.erb]: We can use avoid old hash rocket syntax, same for other fines also.
```suggestion
    <%= f.password_field :password_confirmation, autocomplete: "new-password", class: "form-control", required: true, minlength: 6 %>
```

File: app/views/shared/_flash_messages.html.erb [hunk 2]
```diff
@@ -1,2 +1,6 @@
-<p class="notice"><%= notice %></p>
-<p class="alert"><%= alert %></p>
+<% if notice %>
```
Comment (junan) [inline app/views/shared/_flash_messages.html.erb:1]: Indentation
```erb
<% if notice %>
  <p class="alert alert-success mt-3"><%= notice %></p>
<% end %>
<% if alert %>
  <p class="alert alert-danger mt-3"><%= alert %></p>
<% end %>

```

File: app/views/devise/registrations/new.html.erb [hunk 3]
```diff
@@ -1,29 +1,34 @@
...
+
+  <h2>Sign up</h2>
 
-<%= form_for(resource, as: resource_name, url: registration_path(resource_name)) do |f| %>
   <%= render "devise/shared/error_messages", resource: resource %>
 
-  <div class="field">
-    <%= f.label :email %><br />
-    <%= f.email_field :email, autofocus: true, autocomplete: "email" %>
+  <%= form_for(resource, as: resource_name, url: registration_path(resource_name)) do |f| %>
```
Comment (junan) [inline app/views/devise/registrations/new.html.erb:7]: Indentations

```erb
  <%= form_for(resource, as: resource_name, url: registration_path(resource_name)) do |f| %>
    <div class="form-group">
      <%= f.label :email %>
      <%= f.email_field :email, autofocus: true, autocomplete: "email", class: "form-control", :required => true %>
    </div>

    <div class="form-group">
      <%= f.label :password %>
      <% if @minimum_password_length %>
        <em>(<%= @minimum_password_length %> characters minimum)</em>
      <% end %>
      <%= f.password_field :password, autocomplete: "new-password", class: "form-control", :required => true, :minlength => 6 %>
    </div>

    <div class="form-group">
      <%= f.label :password_confirmation %>
      <%= f.password_field :password_confirmation, autocomplete: "new-password", class: "form-control", :required => true, :minlength => 6 %>
    </div>

    <div class="form-group">
      <%= f.submit "Sign up", class: "form-control btn btn-primary" %>
    </div>
  <% end %>
```

**Review by junan — APPROVED** [review]

File: app/views/shared/_flash_messages.html.erb (hunk 2 above)
Comment (suho) [reply app/views/shared/_flash_messages.html.erb:1]: @junan Nice catch! What IDE are you using? I'm using VSCode + Beautify to format, seemed like the Beautify tool format `.erb` with a wrong indentation config?

Comment (junan) [reply app/views/shared/_flash_messages.html.erb:1]: @suho For Ruby project, I use [Rubymine](https://www.jetbrains.com/ruby/)

**Review by ankitkalia1195 — APPROVED** [review]

File: app/views/devise/registrations/new.html.erb (hunk 1 above)
Comment (suho) [reply app/views/devise/registrations/new.html.erb]: Fixed in [`9a7d980` (#31)](https://github.com/suho/google-search-ruby/pull/31/commits/9a7d9809a136ebbf8cd9847aa9030614cfe2905c)

File: app/views/devise/registrations/new.html.erb (hunk 3 above)
Comment (suho) [reply app/views/devise/registrations/new.html.erb:7]: Fixed in [`5516c98` (#31)](https://github.com/suho/google-search-ruby/pull/31/commits/5516c985d08f72d9f9e17220cbc82f88ee49cb30)

File: app/views/shared/_flash_messages.html.erb (hunk 2 above)
Comment (suho) [reply app/views/shared/_flash_messages.html.erb:1]: Fixed in [`5516c98` (#31)](https://github.com/suho/google-search-ruby/pull/31/commits/5516c985d08f72d9f9e17220cbc82f88ee49cb30)

**Pull request merged**